### ISSUE TITLE
Bug: using Default with imports selects first matching func by name

### ISSUE THREAD

**Issue by na4ma4** [magefile/mage#508]
**Bug Description**
When using imports, namespaces and `Default`, the [getFunction()](https://github.com/magefile/mage/blob/2385abb49a1fa8f1e160556889f4ba9704a2ddb7/parse/parse.go#L642) function will return the first matching function in the import with the same name, regardless of namespacing.

So 

```golang
// project magefile.go
var Default = mage.Test

// imported repo

type Docker mg.Namespace

func (Docker) Test() {
    log.Println("docker")
}

func Test() {
    log.Println("test")
}
```

Will find `Docker.Test()` before `Test()`.

**What did you do?**

Created a [test repository](https://github.com/na4ma4/magefile-test-import) as an example

**What did you expect to happen?**

```shell
$ mage
test
```

**What actually happened?**

```shell
$ mage
docker
```

**Environment**
 - Mage Version: Mage Build Tool 1.15.0
 - OS: darwin/arm64

**Additional context**

**Comment by natefinch** [comment]
ooh, thanks for the report and the repro steps

**Comment by na4ma4** [comment]
I did a PR as well to fix it #509
